openedx/core/djangoapps/catalog/utils.py

```diff
@@ -186,13 +217,8 @@ def get_currency_data():
         list of dict, representing program types.
         dict, if a specific program type is requested.
     """
-    catalog_integration = CatalogIntegration.current()
-    if catalog_integration.enabled:
-        try:
-            user = catalog_integration.get_service_user()
-        except ObjectDoesNotExist:
-            return []
-
+    user, catalog_integration = check_catalog_integration_and_get_user(error_message_field='Currency data')
+    if user:
         api = create_catalog_api_client(user)
```

Comment:
before your change, if user was None (after calling catalog_integration.get_service_user()), this line (and the 2 following) would still execute, but now we are skipping over them when there's no user.  Is that change intentional/correct?

Reply:
I don't believe `catalog_integration.get_service_user()` can return None for the user. It either returns a User object or raises ObjectDoesNotExist.
```
def get_service_user(self):
        # NOTE: We load the user model here to avoid issues at startup time that result from the hacks
        # in lms/startup.py.
        User = get_user_model()  # pylint: disable=invalid-name
        return User.objects.get(username=self.service_username)
```
So I don't think I have changed any current functionality.